src/uct/ib/rc/base/rc_iface.c

```diff
@@ -1055,6 +1047,24 @@ void uct_rc_iface_vfs_refresh(uct_iface_h iface)
     }
 }
 
+void
+uct_rc_iface_adjust_max_get_zcopy(uct_rc_iface_t *iface,
```

Comment:
Would it be easier and more explicit if this function just returns a value instead of internal assignment?
```C
size_t uct_rc_iface_get_max_get_zcopy(...)

iface->config.max_get_zcopy = uct_rc_iface_get_max_get_zcopy(...);
```

Reply:
then need to duplicate `warn(...)` since assignment becomes outside this function

Reply:
I think `warn` can still be inside the function:
```C
{
    if (config->tx.max_get_zcopy == UCS_MEMUNITS_AUTO) {
        return max_tl_get_zcopy;
    } else if (config->tx.max_get_zcopy <= max_tl_get_zcopy) {
        return config->tx.max_get_zcopy;
    } else {
        ucs_warn("%s_iface on %s:%d: reduced max_get_zcopy to %zu",
                 tl_name, dev_name, iface->super.config.port_num,
                 max_tl_get_zcopy);
        return max_tl_get_zcopy;
    }
}
```

Reply:
it's too early to warn since the value is not changed in this scope, we cannot predict caller's behavior